source/develop/release-management/features/virt/parallel-migration-connections.md

```diff
@@ -0,0 +1,88 @@
+
+- Due to their better implementation, parallel migration connections can be faster than the traditional migration implementation even when the network bandwidth can be saturated with a single connection.  However, when single connection is specified, the parallel migration connection implementation uses a different code path, which may not be sufficiently robust, and it's recommended to always use at least two connections.  In the worst case, one of the threads will be idle while causing only a little overhead. 
+
+- Parallel migration connections haven't been tested extensively with connections faster than 100 Gbps.  Due to current multifd implementation, there is an increasing overhead when using many connections and it's not clear whether using more than 16 parallel connections adds a real benefit.
+
+- It still applies that large guests with terrabytes of RAM are difficult to migrate successfully.
+
+- Since it's easier to saturate network bandwidth with multiple connections, it's advisable to not migrate multiple VMs from a single host at the same time.
+
+- VM CPU pinning may restrict the number of CPUs available to the QEMU process and thus impose a limit on the meaningful number of parallel connections.
+
+- It's difficult to guess the right number of parallel connections to use.  A rule of thumb is to use one connection per 10 Gbps of available network bandwidth.  If the number of connections is a bit high, some threads may remain idle while the related overhead should be negligible.
+
+- There is very little documentation about parallel migration connections and the information above was obtained directly from platform developers.
+
+## User experience
+
+A new migration option to enable parallel migration connections is introduced in cluster and VM edit dialogs.  The cluster option specifies what to use in the given cluster.  VMs can override the cluster value individually.  The new option offers the following choices:
+
+- **Disabled** (the default in clusters): Parallel migration connections are not used.
+
+- **Auto**: Use parallel migration connections (TODO: Even when there is insufficient network bandwidth, for the supposed benefit of a more efficient implementation?) and set their number automatically based on the available resources (network bandwidth and CPU power).
+
+- **Custom**: Use parallel migration connections and use the number of connections specified by the user.  The minimum number of connection is 2.  (TODO: Should we put any real limit on the maximum number?)
+
+- **Cluster value** (only in the VM edit dialog; it's the default there): Use parallel connections as specified in the VM's cluster settings. 
+
+We use **Disabled** as the cluster default because it's a newly introduced feature in oVirt and there is not enough experience with it.  We may consider changing it to **Auto** in next releases, especially if QEMU decides using parallel migration connections by default.
```

Comment:
yes and we should also limit that to the new cluster level (that will be added!)

Reply:
OK.